tests/models/llama/test_tokenization_llama.py

```diff
@@ -362,18 +380,56 @@ def test_simple_encode_decode(self):
         self.assertEqual(pyth_tokenizer.encode(" Hello"), [1, 29871, 15043])
         self.assertEqual(rust_tokenizer.encode(" Hello"), [1, 29871, 15043])
 
+    def test_differences_showcase(self):
+        pyth_tokenizer = self.tokenizer
+        rust_tokenizer = self.rust_tokenizer
+        # These are known differences, currently the goal is to *not* get 100% support
+        # since these are edgecases. However, we do document them for future generations.
+        # `tokenizers.Metaspace` with `add_prefix_space` adds a prefix space, Only when
+        # the current pretoken is not already prefixed. This is not how spm does it at
+        # this time (23/03/2023), it ALWAYS forcefully adds a prefix space for the
+        # entire string.
+        # Options to consider in order to implement:
+        #  - Change `add_prefix_space` to ternary, False, True, "force", "force" being
+        # the new version which always prefixes
+        #  - Add a new extra pre_tokenizer which doesn't pretokenize but does this job.
+        #  - Create an entirely new `Metaspace2` which removes options like `split` and `add_prefix_space`
+        #  and try an make everything more composable instead.
+        self.assertEqual(pyth_tokenizer.encode(""), [1])
+        self.assertEqual(rust_tokenizer.encode(""), [1])
+
+        self.assertEqual(pyth_tokenizer.encode(" "), [1, 259])
+        self.assertEqual(rust_tokenizer.encode(" "), [1, 259])
+
+        self.assertEqual(pyth_tokenizer.encode("  "), [1, 1678])
+        self.assertEqual(rust_tokenizer.encode("  "), [1, 1678])
+
+        self.assertEqual(pyth_tokenizer.encode(" Hello"), [1, 29871, 15043])
+        self.assertEqual(rust_tokenizer.encode(" Hello"), [1, 29871, 15043])
+
         self.assertEqual(pyth_tokenizer.encode("<s>"), [1, 1])
         self.assertEqual(rust_tokenizer.encode("<s>"), [1, 1])
 
-        self.assertEqual(pyth_tokenizer.encode(""), [1])
-        self.assertEqual(rust_tokenizer.encode(""), [1])
+    def test_differences_decode(self):
+        pyth_tokenizer = self.tokenizer
+        rust_tokenizer = self.rust_tokenizer
 
         self.assertEqual(pyth_tokenizer.decode([869]), ".")
         self.assertEqual(rust_tokenizer.decode([869]), ".")
 
         self.assertEqual(pyth_tokenizer.decode([30112, 869]), "ا .")
         self.assertEqual(rust_tokenizer.decode([30112, 869]), "ا .")
 
+    def test_differences_special_tokens(self):
```

Comment:
they are no longer different, the merge with the main file was a bit wrong

Reply:
Done.